File: java-api/src/sneer/impl/keys/Keys.java
package sneer.impl.keys;

import sneer.*;

public class Keys {

	public static PrivateKey createPrivateKey() {
		return new PrivateKeyImpl();
	}

	
	public static PrivateKey createPrivateKey(byte[] bytes) {
		return new PrivateKeyImpl(bytes);
	}
	
	
	public static Object createPrivateKey(String bytesAsString) {
		return new PrivateKeyImpl(bytesAsString);
	}
	
	
	public static PublicKey createPublicKey(byte[] bytes) {
		return new PublicKeyImpl(bytes);
	}

	
	public static PublicKey createPublicKey(String bytesAsString) {
		return new PublicKeyImpl(bytesAsString);
	}

}

File: core/src/main/clojure/sneer/keys.clj
(ns sneer.keys
  (:import [sneer.crypto.impl KeysImpl]))

(def ^:private keys-impl (KeysImpl.))

(defn create-prik []
  (.createPrivateKey keys-impl))

(defn create-puk [^bytes rep]
  (.createPublicKey keys-impl rep))

(defn from-hex [^String hex]
  (.createPublicKey keys-impl hex))

(defn ->puk [^String rep]
  (create-puk (.getBytes rep)))
